cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -3551,4 +3618,53 @@ public boolean isSubscriptionChanged() {
   public void setSubscriptionChanged(boolean subscriptionChanged) {
     isSubscriptionChanged = subscriptionChanged;
   }
+
+  /**
+   * This method used for TCF2 CMP
+   * If get consent 1 in IABTCF_VendorConsents at position 1139 then perform subscribe or tracking according to IabTcfMode
+   */
+  private void setTCF() {
+    if (getIabTcfMode() == IabTcfMode.TRACKING_WAIT_FOR_CONSENT) {
+      setTrackingConsentRequired(true);
+    }
+    if (getIabTcfMode() == IabTcfMode.SUBSCRIBE_WAIT_FOR_CONSENT) {
+      setSubscribeConsentRequired(true);
+    }
+    Context mContext = context.getApplicationContext();
+    SharedPreferences mPreferences = PreferenceManager.getDefaultSharedPreferences(mContext);
+
+    SharedPreferences.OnSharedPreferenceChangeListener mListener;
+    mListener = (preferences, key) -> {
+      if (key.equals(IABTCF_VendorConsents)) {
```

Comment:
some early exits in here would be nice e.g.
if (!key.equals(IABTCF_VendorConsents)) {
Logger.d(LOG_TAG, "Vendor consents string is too short to get character at index " + IABTCF_VendorConsent_POSITION + ".");
return;
}

Reply:
In OnSharedPreferenceChangeListener if the preference key is equal to IABTCF_VendorConsents then only it will go inside the if condition otherwise not so it is an early exit. 
As you see we are not doing anything after the if condition.

The OnSharedPreferenceChangeListener works by providing a callback (onSharedPreferenceChanged()) that gets invoked when there is a change in the SharedPreferences. In this case, it listens specifically for changes in the IABTCF_VendorConsents key. When a change occurs, the provided lambda expression is executed, performing the necessary actions based on the updated preferences.

<img width="1137" alt="Screenshot 2023-11-26 at 6 27 22 PM" src="https://github.com/cleverpush/cleverpush-android-sdk/assets/129381063/ec163e53-5057-4166-914a-82c968589056">